core/src/main/java/com/graphhopper/routing/profiles/TagParserFactory.java

```diff
@@ -0,0 +1,495 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.profiles;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.util.AbstractFlagEncoder;
+import com.graphhopper.routing.util.spatialrules.SpatialRule;
+import com.graphhopper.routing.util.spatialrules.SpatialRuleLookup;
+import com.graphhopper.storage.IntsRef;
+import com.graphhopper.util.Helper;
+import com.graphhopper.util.shapes.GHPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class TagParserFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TagParserFactory.class);
+    public static final ReaderWayFilter ACCEPT_IF_HIGHWAY = new ReaderWayFilter() {
+        @Override
+        public boolean accept(ReaderWay way) {
+            return way.getTag("highway") != null;
+        }
+    };
+
+    private static final Set<String> fwdOneways = new HashSet<String>() {
+        {
+            add("yes");
+            add("true");
+            add("1");
+        }
+    };
+
+    public static final String ROUNDABOUT = "roundabout";
+    public static final String ROAD_CLASS = "road_class";
+    public static final String ROAD_ENVIRONMENT = "road_environment";
+    public static final String SURFACE = "surface";
+    public static final String MAX_HEIGHT = "max_height";
+    public static final String MAX_WEIGHT = "max_weight";
+    public static final String MAX_WIDTH = "max_width";
+    public static final String SPATIAL_RULE_ID = "spatial_rule_id";
+    public static final String CURVATURE = "curvature";
+    public static final String CAR_ACCESS = "car.access";
+    public static final String CAR_MAX_SPEED = "car.max_speed";
+    public static final String CAR_AVERAGE_SPEED = "car.average_speed";
+    public static final String BIKE_ACCESS = "bike.access";
+    public static final String BIKE_AVERAGE_SPEED = "bike.average_speed";
+    public static final String FOOT_ACCESS = "foot.access";
+    public static final String FOOT_AVERAGE_SPEED = "foot.average_speed";
```

Comment:
Would it be better to have these keys in a different class? When using them at a later stage, by referencing `TagParserFactory.BIKE_ACCESS`, I don't want to create a new TagParser but use an existing EncodedValue. 

For example 
`this.averageSpeed = em.getDecimalEncodedValue(TagParserFactory.CAR_AVERAGE_SPEED);` in FastestCarWeighting.

At this point I want to read average speeds and not create anything, but I have to use the Factory to get the key.

Reply:
Yes the place is incorrect especially since the keys belong to the core and the TagParserFactory to the `osm` module. Probably Parameters would be a better place but let us first solve the harder problems and write this down as a todo item.